lib/slurp/logs/log_fetcher.ex

```diff
@@ -134,6 +144,10 @@ defmodule Slurp.Logs.LogFetcher do
       toBlock: to_hex_block
     }
 
+    if length(state.address) > 0 do
+      Map.put(filter, :address, state.address)
```

Comment:
This won't update the filter as the return value is discarded. How do you feel about extracting a function to build the filter? i.e.

```suggestion
   defp build_filter(from_block, to_block, state) do
      {:ok, from_hex_block} = from_block |> Slurp.Utils.integer_to_hex()
      {:ok, to_hex_block} = to_block |> Slurp.Utils.integer_to_hex()

      # TODO: Support decamelized keys
      %{
        topics: [state.topics],
        fromBlock: from_hex_block,
        toBlock: to_hex_block
      }
      |> build_filter(state)
    end
    defp build_filter(filter, state) when length(state.address) > 0, do: Map.put(filter, :address, state.address)
    defp build_filter(filter, _), do: filter
```